pkg/storegateway/indexheader/stream_binary_reader_test.go

```diff
@@ -0,0 +1,126 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+
+package indexheader
+
+import (
+	"context"
+	"fmt"
+	"path/filepath"
+	"strings"
+	"testing"
+
+	"github.com/go-kit/log"
+	"github.com/prometheus/prometheus/model/labels"
+	"github.com/prometheus/prometheus/tsdb/fileutil"
+	"github.com/stretchr/testify/require"
+
+	"github.com/thanos-io/objstore/providers/filesystem"
+
+	"github.com/grafana/mimir/pkg/storage/tsdb/block"
+)
+
+type inMemoryLogger struct {
```

Comment:
Instead of testing that particular log statements are emitted, the code in question should be refactored such that it can be tested more easily. This test is in the same package as `stream_binary_reader.go` so you can call unexported methods from the tests if that makes things easier.